### ISSUE TITLE
Support for compressed NES ROMs

### ISSUE THREAD

**Issue by zeusz4u** [ClusterM/hakchi2#433]
I was wondering why retoarch doesn't support ROM images compressed with ZIP. Can anything be done about it? It automatically uncompresses my games when I add them to hackchi... 

Otherwise the whole noIntro 1G1R collection would fit onto it.

**Comment by ClusterM** [comment]
But it support it, as well as .7z...

**Comment by zeusz4u** [comment]
Yeah, but as far as I see it will un-compress the game ROM images when you add them to Hakchi - it doesn't keep them in compressed format.

For example for **MegaMan 6** it created this folder under games: CLV-H-ADERW, which contains the NES rom with **.nes** extension (524,304 bytes). Checked the source I was using (the NoIntro collection), there the ROM was compressed with .zip (258,974 bytes), it's clear it only takes half the disk space when using the compressed ROMs.

I have the Settings -> **Compress non-NES games (for additional emulator mod)** enabled, but this one looks to be applied only for non-NES games. 

Could this compression be activated for NES games as well?

**Comment by zeusz4u** [comment]
Figured out I could keep them and upload them compressed as long as I double-compress them for example: **Mario.nes (original)-> Mario.7z (7zipped) -> Mario.z7.zip (zipped)** , Hakchi would remove the last compression level (zip), and retain the 7z compression. As a trade-off I've also discovered this would remove game metadata from Hakchi and the miniNes (publisher, year, 1/2players is no longer available for compressed ROMs). Also requires fceumm or nestopia emulator core:

https://www.reddit.com/r/nesclassicmods/comments/648uoq/easier_compression_for_nes_games/

Would it be possible to add such a feature in Hakchi, so that you woudn't have to do this for all your games manually, and instead enable an options in Hakchi to keep them compressed on the NES Classic Mini, and also possibly display game metadata for compressed games which will be removed by this move? Thanks
